WIP feat(examples): add p/moul/pageable

moul opened the pull request:
- [ ] Add generic `p/moul/pageable`
- [ ] Refactor `p/demo/avl/pager`
- [ ] Add `p/moul/ulist/pager`
- [ ] Add some wrappers around slices (and maps?)
- [ ] Think more about alternatives, like "cursors" instead of "offsets"

Depends on #3407

moul committed:
feat(examples): add moul/ulist

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
feat(examples): add moul/pageable

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 The pull request was created from a fork (head branch repo: moul/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

moul committed:
Merge branch 'master' into dev/moul/pageable

moul committed:
Merge branch 'master' into dev/moul/pageable

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

n2p5 committed:
Merge branch 'master' into dev/moul/pageable

n2p5 commented on examples/gno.land/p/moul/pageable/pageable.gno:
```diff
@@ -0,0 +1,279 @@
+	totalItems := p.Source.Size()
+	totalPages := 0
+	if pageSize > 0 {
+		totalPages = int(math.Ceil(float64(totalItems) / float64(pageSize)))
```
This can be more simply implemented as

```go
totalPages = (totalItems + pageSize - 1) / pageSize
```

This avoids the need for type conversions.

n2p5 reviewed (changes requested):
I know this is still in draft, but I took a look over the code and it is looking really nice. I put in a couple of notes and pointed out one place that you could avoid type conversions with a fun math trick. Let me know when its out of draft. I like this structure a lot.

n2p5 committed:
Merge branch 'master' into dev/moul/pageable

moul commented:
Related with https://github.com/gnolang/gno/pull/3806
TODO: link to @jeronimoalbi's alternative

moul committed:
Merge remote-tracking branch 'origin/master' into dev/moul/pageable

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
Merge remote-tracking branch 'origin/master' into dev/moul/pageable

# Conflicts:
#	examples/gno.land/p/nt/avl/pager/pager.gno
#	examples/gno.land/p/nt/avl/pager/pager_test.gno
#	examples/gno.land/p/nt/avl/pager/z_filetest.gno

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

jeronimoalbi commented on examples/gno.land/p/moul/pageable/pageable.gno:
```diff
@@ -0,0 +1,301 @@
+//   - Markdown-based pagination UI
+//
+// This package is currently used by:
+//   - gno.land/p/demo/avl/pager: implementation for AVL trees.
```
```suggestion
//   - gno.land/p/nt/avl/pager: implementation for AVL trees.
```

moul committed:
Update examples/gno.land/p/moul/pageable/pageable.gno

Co-authored-by: Jerónimo Albi <894299+jeronimoalbi@users.noreply.github.com>

moul committed:
Update examples/gno.land/p/moul/pageable/pageable.gno

Co-authored-by: Jerónimo Albi <894299+jeronimoalbi@users.noreply.github.com>

moul committed:
Merge branch 'master' into dev/moul/pageable

moul committed:
Merge branch 'master' into dev/moul/pageable

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

jaekwon committed:
Create tictac.md

Do not remove this file.
It is important information, and a canary file.

jaekwon committed:
Added file that proves Reddit censors truthful physics discoveries.

Of all places, /r/UFOResearch should not censor this kind of information.
DO NOT DELETE THIS FILE, DO NOT TRUST ANYONE WHO DELETES THIS FILE.

jaekwon committed:
Update tictac.md

added law of spinning mass.

moul committed:
fix(params): SetBytes should not alter the value bytes (alternative) (#4926)

Alternative for #4905 

Kept:
- Removed GetRaw/SetRaw since they were unused.
- SetBytes to use directly the store, preventing the value to be
aminoJSON encoded.

Updated:
- Preserve API consistency.

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Thomas Bruyelle <thomas.bruyelle@tendermint.com>

jeronimoalbi committed:
feat(examples): simple Markdown forms package (#4977)

Package aims to be a very simple package to define Gno-Flavored Markdown
forms.

It should be easy to update when new form elements or attributes are
introduced.

---------

Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

thehowl committed:
fix(vm/keeper): pre-load stdlib type check cache (#4925)

longer term fix for #4874, re-introducing the type check cache for
stdlibs.

jeronimoalbi committed:
feat(boards2): change realm to use `gno.land/p/gnoland/boards` (#4939)

PR changes Boards2 realm to use `gno.land/p/gnoland/boards` package
which makes the realm leaner.

It also improves `BasicPermissions` implementation by using a custom
members storage implementation that makes users iteration cheaper.

mvallenet committed:
fix(vm/keeper): improve error for non crossing msg call error (#4899)

fix #4432

---------

Co-authored-by: Morgan <morgan@morganbaz.com>

moul committed:
test: add a banker security txtar (#4987)

Signed-off-by: moul <94029+moul@users.noreply.github.com>

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

jeronimoalbi committed:
feat(boards2): open boards support (#4993)

jeronimoalbi committed:
feat(boards2): support Markdown forms (#4996)

jeronimoalbi committed:
chore(boards2): enable open boards field in create board form (#4998)

Also add a couple of minor fixes

jaekwon committed:
first commit of whitepaper; draft; DO NOT SHARE, deadman switch

moonia committed:
fix(gnoweb): handle multiline value for gno-textarea in forms (#4978)

Closes https://github.com/gnolang/gno/issues/4968

## Description

The `<gno-textarea>` component was literally displaying the `\n`
sequence. After fix, the parser now interprets the `\n` sequence as a
real newline character.

### Example

```
<gno-form>
  <gno-textarea name="content" value="         First line\nSecond line" />
</gno-form>
```

### Before fix: 
<img width="931" height="236" alt="Capture d’écran 2025-12-15 à 23 22
23"
src="https://github.com/user-attachments/assets/220f85ee-c5ad-4dd3-9b26-863bd84e6a58"
/>


### After fix: 
<img width="945" height="239" alt="Capture d’écran 2025-12-15 à 23 34
27"
src="https://github.com/user-attachments/assets/e8f7551e-749f-4259-ba50-5dae333cf487"
/>

---------

Co-authored-by: alexiscolin <alexis@jaunebleu.co>

moul committed:
Merge master into dev/moul/pageable

thehowl commented:
Closing as 1.5yr stale
